### PR TITLE
Syntax highlight codeblocks

### PR DESCRIPTION
**Pull request by jessp01**
- Support syntax colour highlighting with `gohighlight`
 - Support the option to add a new page when `HorizontalRule` is encountered

### CODE REVIEW

File: .gitmodules
Comment (mandolyte) [inline .gitmodules:1]: @jessp01 I didn't spot any way to test your changes. So I created a markdown file like this:

```
$ cat test_syntax_highlighting.md
# Test Syntax Highlighting for Code Blocks

This is a Go code block

```go
package main

func main() {
        println("Hello World")
}
```

This is a Javascript code block

```js
function myfunc() {
        console.log("hello world");
}
```

Then ran: `go run convert.go -i test_syntax_highlighting.md -o test_syntax_highlighting.pdf`

With this result:

image

I was expecting something that highlighted the language keywords. Am I correct? Is there something else that needs to be done?

Comment (jessp01) [reply .gitmodules:1]: Hi @mandolyte ,

Please see the section I added  to the README (https://github.com/mandolyte/mdtopdf/pull/30/commits/63a9c11b4656aea22bec992d869a63b5cf2e6e74)
> To benefit from Syntax highlighting, invoke thusly:
go run convert.go -i syn_test.md -s /path/to/syntax_files -o test.pdf

I also added a test case https://github.com/mandolyte/mdtopdf/blob/99c7fc99aa8c118b43d8b51e5e40bce12b03bf02/mdtopdf_test.go#L69 and committed the soruce and result (https://github.com/mandolyte/mdtopdf/blob/99c7fc99aa8c118b43d8b51e5e40bce12b03bf02/testdata/Markdown%20Documentation%20-%20Colour.pdf)

**Review by mandolyte — APPROVED** [review]
Thanks for this PR. Very nice. After playing with it I'll probably add an example to my "cmd" folder and update the readme or provide some other way to include details of what a user must have on hand to make use of the syntax highlighting.

**Comment by jessp01** [comment]
Hi @mandolyte ,

Glad you like it:)
I created a pull with a small adjustment, see https://github.com/mandolyte/mdtopdf/pull/31.

As to adding another example, of course, feel free to add whatever you feel is right but I should note that the current `./cmd/convert.go` can also be altered such that the default creation of `pf` becomes:

```go
            pf = mdtopdf.NewPdfRenderer("", "", *output, "trace.log",
                    mdtopdf.SetSyntaxHighlightBaseDir("../highlight/syntax_files"))
```

This will work correctly because I added `gohighlight` as a submodule.

**Comment by jessp01** [comment]
Also, following your comments in this commit: https://github.com/mandolyte/mdtopdf/commit/e361b3dcca804a186209fea1b28db96e300f5d1a, see my commit here: https://github.com/jessp01/gohighlight/commit/8bad7b78d45adec360c9d1d838724ca32b6bafd3

With this symlink in place 
```js
console.log("hello work")
```

Will also work. To update the submodule, just run `git submodule update --remote` from your clone and push:)

**Pull request merged**